Game/Addons/IncreasedTerrainDistanceMod/Scripts/IncreasedTerrainDistance.cs

```diff
@@ -0,0 +1,979 @@
+        Material terrainMaterial = null;
+
+        // map holds info for tiles of terrain: r-channel... climate index, gb... currently unused, a-channel... discard-rendering flag for shader
+        Color32[] terrainInfoTileMap = null;
+        int terrainInfoTileMapDim;
+
+        // texture for terrainInfoTileMap
+        Texture2D textureTerrainInfoTileMap = null;
+
+
+        ClimateSeason oldSeason = ClimateSeason.Summer;
+
+        Texture2D textureAtlasDesertSummer = null;
+        Texture2D textureAtlasWoodlandSummer = null;
+        Texture2D textureAtlasMountainSummer = null;
+        Texture2D textureAtlasSwampSummer = null;
+
+        Texture2D textureAtlasDesertWinter = null;
+        Texture2D textureAtlasWoodlandWinter = null;
+        Texture2D textureAtlasMountainWinter = null;
+        Texture2D textureAtlasSwampWinter = null;
+
+        Texture2D textureAtlasDesertRain = null;
+        Texture2D textureAtlasWoodlandRain = null;
+        Texture2D textureAtlasMountainRain = null;
+        Texture2D textureAtlasSwampRain = null;
+
+        // stacked near camera (used for near terrain from range 1000-15000) to prevent floating-point rendering precision problems for huge clipping ranges
+        Camera stackedNearCamera = null; 
+
+        // stacked camera (used for far terrain) to prevent floating-point rendering precision problems for huge clipping ranges
+        Camera stackedCamera = null;
+
+        public Camera getFarTerrainCamera() { return stackedCamera; }
+        public Camera getStackedNearCamera() { return stackedNearCamera; }
+
+        #if ENHANCED_SKY_AVAILABLE
+        EnhancedSky.SkyManager skyMan = null;
+        bool sampleFogColorFromSky = false;
+        #endif
+
+        GameObject goRenderSkyboxToTexture = null;
+        Camera cameraRenderSkyboxToTexture = null;
+
+        const int renderTextureSkyWidth = 256;
+        const int renderTextureSkyHeight = 256;
+        const int renderTextureSkyDepth = 16;
+        const RenderTextureFormat renderTextureSkyFormat = RenderTextureFormat.ARGB32;
+        RenderTexture renderTextureSky = null;
+
+
+        // instance of dfUnity
+        DaggerfallUnity dfUnity;
+
+        #endregion
+
+        #region Properties
+
+        public bool IsReady { get { return ReadyCheck(); } }
+
+        #endregion
+
+        #region Unity
+
+        static string GetGameObjectPath(GameObject obj)
+        {
+            string path = "/" + obj.name;
+            while (obj.transform.parent != null)
+            {
+                obj = obj.transform.parent.gameObject;
+                path = "/" + obj.name + path;
+            }
+            return path;
+        }
+
+        void Awake()
+        {
+            dfUnity = DaggerfallUnity.Instance;
+
+            //DaggerfallUnity.OnSetTerrainSampler += SetTerrainSampler;
+            
+            DaggerfallUnity.Instance.TerrainSampler = new ImprovedTerrainSampler();            
```

Comment:
Please change this to be assigned using event DaggerfallUnity.OnSetTerrainSampler. Also should check if mod is enabled at startup using DaggerfallUnity.Settings.Nystul_IncreasedTerrainDistance flag.